sdk/metrics/src/main/java/io/opentelemetry/sdk/metrics/SdkMeterProvider.java

```diff
@@ -41,10 +35,8 @@ public final class SdkMeterProvider implements MeterProvider, Closeable {
 
   private final ComponentRegistry<SdkMeter> registry;
   private final MeterProviderSharedState sharedState;
-  private final Map<CollectionHandle, CollectionInfo> collectionInfoMap;
+  private final List<RegisteredReader> registeredReaders;
   private final AtomicBoolean isClosed = new AtomicBoolean(false);
-  private final AtomicLong lastCollectionTimestamp;
-  private final long minimumCollectionIntervalNanos;
```

Comment:
The `minimumCollectionIntervalNanos` we've previously talked about turned out to be central to the bug. It's true purpose appears to be trying to ensure that when multiple readers are present they each receive the same data if they collect within a narrow enough interval of time. However, I believe the mechanism to be flawed as it didn't account for correctness when the readers are on different schedules (a perfectly reasonable scenario).

The refactor negates the need for it and assists in reducing complexity.